### DISCUSSION TITLE
New Wyze Cam v3 (PRO) out

### DISCUSSION THREAD

**Discussion by skilau** [gtxaspec/wz_mini_hacks#366] [General]
New Wyze Cam v3 Pro out:
https://www.wyze.com/products/wyze-cam-pro

I am guessing they probably fixed their bootloader error, thus blocking us from installing our own stuff.

Would be interesting to check out.

**Comment by endertable** [comment]
Agreed, i have a few on order.  So will be interesting to see

**Reply by Bob-AOL** [reply]
Bitdefender tells me that link is an unsafe site. LOL

**Reply by endertable** [reply]
I ordered mine through the Wyze app.  You get discounts through the app as well

**Comment by virmaior** [comment]
I noticed 
1. still only 20/15 fps
2. a bit of a weasel wording `Local Storage: Supports up to [256 GB microSD](https://www.wyze.com/products/wyze-128gb-and-32gb-microsdhc-card-class-10) in FAT32 format` since this was already functioning
3. That's a new processor right?

**Reply by endertable** [reply]
Looks like they are holding back on some of the details like the processor.  Guess they figure, the less we know, the harder to attack, hehe.

**Reply by gtxaspec** [reply]
ingenic t40

**Comment by mrlt8** [comment]
We should be seeing an updated Pan Pro cam as well. 

One potential downside is for people running their cams in homekit since the video will need to be re-encoded to HD h264 at some point.

**Comment by UncleJohnsBnd** [comment]
Wondering if the hack will work as well... Along with ability to use wired network adapter.

Anxiously awaiting any reports... 😁

**Reply by foureight84** [reply]
I have a feeling it will ship with newer locked-down firmware.

**Comment by endertable** [comment]
I ordered one on the 24th and today through email found it was a preorder.  :(
Oh well, so new ETA is middle of November.  :)

**Reply by Bob-AOL** [reply]
The ad on their site indicates shipping begins on the 4th.

**Comment by endertable** [comment]
Looks like their online update is one huge 16MB .bin file that seems like it is encrypted in some way.

**Reply by gtxaspec** [reply]
url?

**Reply by endertable** [reply]
https :// s3-us-west-2.amazonaws.com/wuv2/upgrade/HL_CAM3P/firmware/4.58.0.2905.gz

**Reply by skilau** [reply]
7zip extracts it (you have to extract the resulting files a couple of times)
It results in a typical Linux tree of files.  (/bin /etc /lib) etc.

**Reply by kohrar** [reply]
Thanks for that URL @endertable!

I was also able to read the image file. There's a cpio archive and a squashfs filesystem there. The squashfs portion contains the camera's firmware (iCamera's there).

```
$ binwalk image.bin
DECIMAL       HEXADECIMAL     DESCRIPTION
--------------------------------------------------------------------------------
131136        0x20040         Flattened device tree, size: 5078985 bytes, version: 17
131356        0x2011C         Flattened device tree, size: 17209 bytes, version: 17
153288        0x256C8         LZMA compressed data, properties: 0x5D, dictionary size: 67108864 bytes, uncompressed size: -1 bytes
2534564       0x26ACA4        gzip compressed data, maximum compression, from Unix, last modified: 1970-01-01 00:00:00 (null date)
5210124       0x4F800C        Squashfs filesystem, little endian, version 4.0, compression:xz, size: 10786752 bytes, 626 inodes, blocksize: 131072 bytes, created: 2022-10-25 
```

**Comment by gtxaspec** [comment]
amazon sold out of cameras, forgot to preorder :(

**Reply by kohrar** [reply]
Sadly, I don't see them for sale in Canada yet

**Comment by endertable** [comment]
Got it!   :)   Apparently, I had to reinstall binwalk!  :(

But, this looks dangerously hackable.

According to the app, mine got fulfilled today so hopefully soon   :)

**Comment by endertable** [comment]
Received mine today.  Anyone have any ides they want me to try?

**Reply by UncleJohnsBnd** [reply]
First thing would be does the mini hacks load and run.

Second would be if minin hacks run does the usb network support still work.

**Reply by gtxaspec** [reply]
wz_mini will not work, the new camera uses a newer kernel.  stand by

**Reply by UncleJohnsBnd** [reply]
Standing by... 😁

**Reply by endertable** [reply]
I am curious if they decided to go with secure uboot on this one.  I also saw a signature/key file in /system/bin that usually verifies partition integrity.  Not surenof they are using it

**Reply by skilau** [reply]
Definitely interested in how this plays out!  Hopefully root can be had on these!

**Reply by mrlt8** [reply]
Would appreciate it if you could test out the latest version of wyze-bridge!

It seems like the pro is still using h264 for the 2k video stream.

**Comment by netappian** [comment]
Did anyone get wz_mini to work on V3 Pro ?

**Reply by virmaior** [reply]
won't work out of the box due to secure boot. as far as is currently known, you would need to replace the main processor with one that is not locked.

**Reply by b00l3an** [reply]
Shame, swapping processors is probably a bit beyond me. 
Thanks for answering!
